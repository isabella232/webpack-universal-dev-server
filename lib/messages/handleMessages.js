// @flow

import type { ChildProcess } from "child_process";
import type { Message, MessageHandler, MessageHandlers } from "../types";

type ProcessLike = Process | ChildProcess;
type ProcessMessage = Message;
type ProcessMessageHandler = MessageHandler<ProcessMessage>;

const passMessage = (
    message: ProcessMessage,
    handler?: ProcessMessageHandler
): void =>
    void (
        handler && handler(message)
    );
const isValidMessage = (
    message
): boolean =>
    message && typeof message.type === "string";
const handleProcessMessage = (
    process: ProcessLike,
    handlers: MessageHandlers
) => (
    message: ProcessMessage
): void =>
    void (
        isValidMessage(message) &&
        passMessage(
            message,
            handlers.get(message.type)
        )
    );

export default (
    process: ProcessLike,
    handlers: MessageHandlers
): void =>
    void (
        process.on("message", handleProcessMessage(process, handlers))
    );
