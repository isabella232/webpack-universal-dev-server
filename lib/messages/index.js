// @flow

import type {
    WpDevServerPauseMessage, WpDevServerResumeMessage,
    WpCompilationMessage, WpDoneMessage,
    WebpackStats
} from "../types"

export const TYPE_WP_DEV_SERVER_PAUSE = "WP_DEV_SERVER/PAUSE"
export const TYPE_WP_DEV_SERVER_RESUME = "WP_DEV_SERVER/RESUME"
export const TYPE_WP_COMPILATION = "WP/COMPILATION"
export const TYPE_WP_DONE = "WP/DONE"
export const TYPE_APP_SERVER_LISTENING = "APP_SERVER/LISTENING"

export const wpDevServerPause = (
): WpDevServerPauseMessage =>
    ({
        type: TYPE_WP_DEV_SERVER_PAUSE
    })

export const wpDevServerResume = (
): WpDevServerResumeMessage =>
    ({
        type: TYPE_WP_DEV_SERVER_RESUME
    })

export const wpCompilation = (
    publisher: string
): WpCompilationMessage =>
    ({
        type: TYPE_WP_COMPILATION,
        publisher
    })

export const wpDone = (
    publisher: string,
    stats: WebpackStats
): WpDoneMessage =>
    ({
        type: TYPE_WP_DONE,
        publisher,
        duration: stats.endTime - stats.startTime,
        hasErrors: stats.hasErrors(),
        hasWarnings: stats.hasWarnings(),
        errorLog: stats.toString("errors-only")
    })

// appServerListening message is not defined here because the message is created by the application itself
// See also our webpack-universal-tools library
