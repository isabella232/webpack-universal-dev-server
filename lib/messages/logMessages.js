// @flow

import { EOL } from "os";
import chalk from "chalk";

const separator = "---------------------------------------------------------------------------------------";
const finished = (
    what: string
): string =>
    chalk.green(`${ what } finished`);
const formatDuration = (
    duration: number
): string =>
    (duration / 1000).toFixed(1);

export const buildStarted = (
    what: string
): string =>
    `${ chalk.yellow(what) } started...`;

export const buildSuccess = (
    what: string,
    duration: number
): string =>
    `${ finished(what) } after ${ formatDuration(duration) }s`;

export const buildError = (
    what: string,
    error: string
): string =>
    [
        "",
        separator,
        chalk.red(`${ what } error`),
        separator,
        "",
        error,
        "",
        separator,
    ].join(EOL);
