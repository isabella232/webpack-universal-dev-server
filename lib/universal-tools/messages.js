// @flow

import type {
    AppServerListeningMessage,
} from "./types";

export const TYPE_APP_SERVER_LISTENING = "APP_SERVER/LISTENING";

export const appServerListening = (
    port: number,
    host: string = "localhost"
): AppServerListeningMessage =>
    ({
        type: TYPE_APP_SERVER_LISTENING,
        port,
        host,
    });
