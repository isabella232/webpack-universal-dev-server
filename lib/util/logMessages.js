// @flow

import { EOL } from "os";
import chalk from "chalk";

const separator = "---------------------------------------------------------------------------------------";

export function buildStarted(what: string): string {
    return `${ chalk.yellow(what) } started...`;
}

export function buildSuccess(what: string, duration: number): string {
    const green = chalk.green(`${ what } finished`);

    return `${ green } after ${ (duration / 1000).toFixed(1) }s`;
}

export function buildError(what: string, error: string): string {
    return [
        "",
        separator,
        chalk.red(`${ what } error`),
        separator,
        "",
        error,
        "",
        separator
    ].join(EOL);
}
