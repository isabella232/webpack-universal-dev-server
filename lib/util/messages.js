// @flow

const NAMESPACE = __filename;

// WDS = Webpack Dev Server (watches, compiles and serves the client bundle)
// WWP = Webpack Watch Process (watches and compiles the server bundle)
export const TYPE_WDS_PAUSE = `${ NAMESPACE }/WDS/PAUSE`;
export const TYPE_WDS_RESUME = `${ NAMESPACE }/WDS/RESUME`;
export const TYPE_WWP_COMPILATION = `${ NAMESPACE }/WWP/COMPILATION`;
export const TYPE_WWP_DONE = `${ NAMESPACE }/WWP/DONE`;

export type Message = {
    type: string;
};

export function wdsPause(): Message {
    return {
        type: TYPE_WDS_PAUSE
    };
}

export function wdsResume(): Message {
    return {
        type: TYPE_WDS_RESUME
    };
}

export function wwpCompilation(): Message {
    return {
        type: TYPE_WWP_COMPILATION
    };
}

export function wwpDone(): Message {
    return {
        type: TYPE_WWP_DONE
    };
}
