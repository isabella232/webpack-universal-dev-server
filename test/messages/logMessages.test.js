import test from "ava";
import { buildStarted, buildSuccess, buildError } from "../../lib/messages/logMessages";
import normalizeLog from "../helpers/normalizeLog";

function check(message) {
    return t => {
        t.snapshot(normalizeLog(message));
    };
}

test(
    "should log the expected buildStarted",
    check(buildStarted("webpack client build"))
);

test(
    "should log the expected buildSuccess message with durations above 1 second",
    check(buildSuccess("webpack client build", 1300))
);

test(
    "should log the expected buildSuccess message with durations below 1 second",
    check(buildSuccess("webpack client build", 300))
);

test(
    "should log the expected buildError message",
    check(buildError("webpack client build", "Module build failed"))
);
