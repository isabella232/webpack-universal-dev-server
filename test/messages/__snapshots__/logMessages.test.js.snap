exports[`should log the expected buildError message 1`] = `"\"\\n---------------------------------------------------------------------------------------\\n\\u001b[31mwebpack client build error\\u001b[39m\\n---------------------------------------------------------------------------------------\\n\\nModule build failed\\n\\n---------------------------------------------------------------------------------------\""`;

exports[`should log the expected buildStarted 1`] = `"\"\\u001b[33mwebpack client build\\u001b[39m started...\""`;

exports[`should log the expected buildSuccess message with durations above 1 second 1`] = `"\"\\u001b[32mwebpack client build finished\\u001b[39m after 1.3s\""`;

exports[`should log the expected buildSuccess message with durations below 1 second 1`] = `"\"\\u001b[32mwebpack client build finished\\u001b[39m after 0.3s\""`;
